### PR TITLE
[OBSOLETE] [RHO-4290] Consolidate logging interface to require channel to be specified

### PR DESCRIPTION
**Pull request by jpersons-series**
#### Description

Consolidated the log methods such that logging to default without specifying a channel is no longer an option.

Now it looks like:
image

#### Testing plan

Case by case basis, we will need to rollout one-by-one and case by case basis

#### JIRA + Optional Slack context

https://series-ai.atlassian.net/browse/RHO-4290
https://seriesai.slack.com/archives/C04QCDML1AR/p1726674375358839

### CODE REVIEW

File: PadoruLogger/Debug.cs [hunk 1]
```diff
@@ -306,7 +243,7 @@ private static void AutoConfigure()
 
             AddOutput(defaultOutput);
 
-            LogWarning($"Tried to use Padoru.Diagnostics.Debug without configuring it first. Logger auto-configured itself with default options.");
+            LogWarning($"Tried to use Padoru.Diagnostics.Debug without configuring it first. Logger auto-configured itself with default options.", "LogInternal");
```
Comment (kadak4) [inline PadoruLogger/Debug.cs]: This channel should be a constant and called something more informative like `PadoruDebug`

File: PadoruLogger/Debug.cs [hunk 2]
```diff
@@ -99,87 +97,26 @@ public static void AddOutput(IDebugOutput output)
...
-            LogError(message, DEFAULT_CHANNEL_NAME, context);
-        }
-
-        public static void LogError(object message, string channel, object context)
+        public static void LogError(object message, string channel, object context=null)
         {
             InternalLog(LogType.Error, message, channel, context);
         }
 
-        public static void LogException(Exception e)
```
Comment (kadak4) [inline PadoruLogger/Debug.cs:163]: We should keep this overload, no reason to sent null as the messageHeader if we do not want one

File: PadoruLogger/Debug.cs [hunk 3]
```diff
@@ -99,87 +97,26 @@ public static void AddOutput(IDebugOutput output)
...
-        {
-            LogException(null, e, null);
-        }
-
-        public static void LogException(object messageHeader, Exception e)
```
Comment (kadak4) [inline PadoruLogger/Debug.cs:168]: Same here, context is something we rarely have, we should keep this overload

File: PadoruLogger/Debug.cs [hunk 4]
```diff
@@ -99,87 +97,26 @@ public static void AddOutput(IDebugOutput output)
...
-        {
-            LogException(messageHeader, e, null);
-        }
-
-        public static void LogException(Exception e, object context)
```
Comment (kadak4) [inline PadoruLogger/Debug.cs:173]: Same here

File: PadoruLogger/Debug.cs [hunk 5]
```diff
@@ -99,87 +97,26 @@ public static void AddOutput(IDebugOutput output)
...
-        {
-            LogException(null, e, context);
-        }
-        
-        public static void LogException(object messageHeader, Exception e, object context)
```
Comment (kadak4) [inline PadoruLogger/Debug.cs:178]: Same here

**Review by kadak4 — CHANGES_REQUESTED** [review]

**Comment by jpersons-series** [comment]
New PR: https://github.com/series-ai/logger-lib/compare/main...jpersons/RHO-4290_require-log-channel?expand=1

**Pull request closed without merging**